common/index.ts

```diff
@@ -164,3 +164,19 @@ export type Snowflake = `${bigint}`;
  * @internal
  */
 export type Permissions = `${bigint}`;
+
+/**
+ * https://discord.com/developers/docs/reference#message-formatting-formats
+ */
+export const FormattingPatterns = {
+	User: /<@(\d{17,19})>/,
+	UserWithNickname: /<@!(\d{17,19})>/,
+	UserWithOptionalNickname: /<@!?(\d{17,19})>/,
+
+	Channel: /<#(\d{17,19})>/,
+	Role: /<@&(\d{17,19})>/,
+
+	Emoji: /<?(a)?:?(\w{2,32}):(\d{17,19})>?/,
+	AnimatedEmoji: /<(a):(\w{2,32}):(\d{17,19})>?/,
+	StaticEmoji: /<:(\w{2,32}):(\d{17,19})>?/,
```

Comment:
```suggestion
	Emoji: /<(a)?:(\w{2,32}):(\d{17,19})>/,
	AnimatedEmoji: /<(a):(\w{2,32}):(\d{17,19})>/,
	StaticEmoji: /<:(\w{2,32}):(\d{17,19})>/,
```
those aren't optional for the other mention patterns either